### PR TITLE
pleroma-bot: init at 0.8.6

### PR DESCRIPTION
**Pull request by robertoszek**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -8039,6 +8039,12 @@
     githubId = 3708689;
     name = "Roberto Di Remigio";
   };
+  robertoszek = {
+    email = "robertoszek@robertoszek.xyz";
+    github = "robertoszek";
+    githubId = 1080963;
+    name = "Roberto";
+  };
```
Comment (humancalico) [inline maintainers/maintainer-list.nix:8047]: Can you split this in a different commit with the message `maintainers: add robertoszek`?

File: pkgs/development/python-modules/pleroma-bot/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+buildPythonPackage rec {
+  pname = "pleroma-bot";
+  version = "0.8.6";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "85056df457a49f2adbd36349d821c9401003d91a753de6456ceb805770e38c13";
+  };
+
+  doCheck = false;
```
Comment (humancalico) [inline pkgs/development/python-modules/pleroma-bot/default.nix:19]: Can you add a comment with the reason for why checks have to be disabled?

**Review by akshatagarwl — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/pleroma-bot/default.nix (hunk 2 above)
Comment (fabaff) [inline pkgs/development/python-modules/pleroma-bot/default.nix:19]: The upstream repo contains [`tests`](https://github.com/robertoszek/pleroma-bot/tree/master/pleroma_bot/tests). If they are mocking network access properly, perhaps we could run them.

If it's not possible to run the tests, at least a `pythonImportsCheck` should be added.

File: pkgs/development/python-modules/pleroma-bot/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  propagatedBuildInputs = [ pyaml requests requests_oauthlib oauthlib ];
+
+  meta = with lib; {
+    homepage = "https://github.com/robertoszek/pleroma-bot";
```
Comment (fabaff) [inline pkgs/development/python-modules/pleroma-bot/default.nix:23]: The upstream project has a separate documentation website in addition to their GitHub presence.

```suggestion
    homepage = "https://robertoszek.github.io/pleroma-bot/";
```

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (robertoszek) [reply maintainers/maintainer-list.nix:8047]: Done. Apologies for bundling it into the same commit.

File: pkgs/development/python-modules/pleroma-bot/default.nix (hunk 2 above)
Comment (robertoszek) [reply pkgs/development/python-modules/pleroma-bot/default.nix:19]: The archive retrieved by ```fetchPypi``` doesn't include ```tests```, not sure how to go about that.
Would it be possible and/or acceptable to retrieve them some other way in order to run them? (checking the documentation I've not come across any example or mention of retrieving tests from a different source than the package itself).

I've added ```pythonImportsCheck``` in the meantime, as requested.

File: pkgs/development/python-modules/pleroma-bot/default.nix (hunk 3 above)
Comment (robertoszek) [reply pkgs/development/python-modules/pleroma-bot/default.nix]: I wasn't completely sure what the ```homepage``` value should've been in this case. This clears things up, thanks.

File: pkgs/development/python-modules/pleroma-bot/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/python-modules/pleroma-bot/default.nix:19]: Please fetch the repo from github or the hosting platform of their choice instead.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -24041,6 +24041,8 @@ in
      stdenv = stdenv_32bit;
   };
 
+  pleroma-bot = python3Packages.callPackage ../development/python-modules/pleroma-bot {};
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24044]:
```suggestion
  pleroma-bot = python3Packages.callPackage ../development/python-modules/pleroma-bot { };
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash all package related commits into one commit.

File: pkgs/development/python-modules/pleroma-bot/default.nix (hunk 2 above)
Comment (robertoszek) [reply pkgs/development/python-modules/pleroma-bot/default.nix]: Understood, I changed it to ```fetchFromGitHub``` and added the pytest tests. 
The package builds successfully and the tests pass.

Comment (robertoszek) [reply pkgs/development/python-modules/pleroma-bot/default.nix]: I was a bit confused as to what was the best source for the package. If I retrieved the archive from PyPI, no tests were included.
I've now changed it to ```fetchFromGitHub``` and added the pytest tests.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114670` at 312aec13 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/10a9876836710b6cb7cf1b251f27ba57">pleroma-bot</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-explicit-check-phase.md">python-explicit-check-phase</a>

Consider using `pytestCheckHook` in `checkInputs` rather than `checkPhase = "pytest";`.

Near pkgs/development/python-modules/pleroma-bot/default.nix:24:3:
```
   |
24 |   checkPhase = "pytest";
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/pleroma-bot/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner  = "robertoszek";
+    repo   = "pleroma-bot";
+    rev    = "0.8.6";
+    sha256 = "1q0xhgqq41zbqiawpd4kbdx41zhwxxp5ipn1c2rc8d7pjyb5p75w";
+  };
+
+  checkInputs = [ pytest ];
+  checkPhase = "pytest";
```
Comment (fabaff) [inline pkgs/development/python-modules/pleroma-bot/default.nix:24]: Looks like that `pytestCheckHook` could be used.

```suggestion
  checkInputs = [ pytestCheckHook ];
```

Comment (robertoszek) [reply pkgs/development/python-modules/pleroma-bot/default.nix:24]: Yep, I just tried and executes all tests as expected. Thanks for the suggestion

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 114670` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>pleroma-bot</li>
  </ul>
</details>

File: pkgs/development/python-modules/pleroma-bot/default.nix [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  src = fetchFromGitHub {
+    owner  = "robertoszek";
+    repo   = "pleroma-bot";
+    rev    = "0.8.6";
+    sha256 = "1q0xhgqq41zbqiawpd4kbdx41zhwxxp5ipn1c2rc8d7pjyb5p75w";
+  };
+
+  checkInputs = [ pytestCheckHook ];
+
+  buildInputs = [ requests-mock ];
```
Comment (fabaff) [inline pkgs/development/python-modules/pleroma-bot/default.nix:25]: `requests-mock` is a [test requirement](https://github.com/robertoszek/pleroma-bot/blob/master/setup.py#L51).

```suggestion
  checkInputs = [ pytestCheckHook requests-mock ];
```

**Review by fabaff — COMMENTED** [review]
Nice work. We are down to three remaining issues. Two are not blockers.

`nixpkgs-hammering` complain about the ordering. A lot of people place `propagatedBuildInputs` before `checkInputs` as it makes manual checking easier (`propagatedBuildInputs` don't have to be repeated in `checkInputs`). `buildInputs` and `nativeBuildInputs` comes first.

```bash
When evaluating attribute ‘pleroma-bot’:
warning: attribute-ordering
The lists of dependencies, including the attribute “buildInputs” should preferably come before lists of dependencies’ “checkInputs” attribute in the expression.

Near /home/fab/nixpkgs/pkgs/development/python-modules/pleroma-bot/default.nix:23:3:
   |
23 |   checkInputs = [ pytestCheckHook ];
   |   ^
Near /home/fab/nixpkgs/pkgs/development/python-modules/pleroma-bot/default.nix:25:3:
   |
25 |   buildInputs = [ requests-mock ];
   |   ^
See: https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/attribute-ordering.md
```

Please run `nixpkgs-fmt`. `src` seems a little off.

File: pkgs/development/python-modules/pleroma-bot/default.nix (hunk 6 above)
Comment (robertoszek) [reply pkgs/development/python-modules/pleroma-bot/default.nix]: Thanks for the explanation, your guidance has been immensely helpful 👍

File: pkgs/development/python-modules/pleroma-bot/default.nix [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+}:
+
+buildPythonPackage rec {
+  pname = "pleroma-bot";
+  version = "0.8.6";
+
+  src = fetchFromGitHub {
+    owner = "robertoszek";
+    repo = "pleroma-bot";
+    rev = "0.8.6";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pleroma-bot/default.nix:19]:
```suggestion
    rev = version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114670` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>1 package failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>pleroma-bot: <a href="https://termbin.com/zle5">plain log</a>  |  <a href="https://htmlpreview.github.io/?https://gist.githubusercontent.com/SuperSandroBot/71164d98854b339a9082262051e4e6b2/raw/index.html">pretty log </a></li>
  </ul>
</details>
